### ISSUE TITLE
Implement services to interact with the REST API

### ISSUE THREAD

**Issue by artlowel** [DSpace/dspace-angular#31]
This task depends on #30 

- For the Collections, Items and Bitstreams endpoints, create a service, a reducer an actions and an effects file
- each service should have
	- a findById method
	- a findAll method, that supports pagination and sorting
	- a findAllInScope method, and supports pagination and sorting
- when a find method is called, dispatch the appropriate FIND_REQUEST action (e.g. `dspace/shared/data/item/FIND_BY_ID_REQUEST`)
- in the corresponding effects file, listen for that FIND_REQUEST action
	- when it arrives, call the appropriate endpoint using ApiService
	- If the call succeeds, use a JSONAPISerializer to deserialize the result, and dispatch the appropriate FIND_SUCCESS action that adds the object(s) to the store
	- If the response is an error, or times out, dispatch the appropriate FIND_ERROR action with an error message.
- The find methods should return observables that select the proper values from the store
- You can find a simplified example of this way of working in [this test branch](https://github.com/DSpace/dspace-angular/compare/master...artlowel:yayson-test?diff=split&name=yayson-test)
- Don't worry about caching at the moment, that's part of a separate task.

**Comment by artlowel** [comment]
The bulk of the code for this issue has been merged with #62
To do: more tests and docs

**Comment by artlowel** [comment]
This ticket was left open because the code still lacks tests and documentation. This was moved to a separate issue: #130
